core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -313,6 +303,33 @@ final class ShuffleBlockFetcherIterator(
   }
 }
 
+// Helper class that ensures a ManagedBuffer is released on InputStream.close()
+private class WrappedInputStream(
+              delegate: InputStream, buf: ManagedBuffer, var currentResult: FetchResult)
```

Comment:
Style nit: the parameters here need to be wrapped differently; see https://cwiki.apache.org/confluence/display/SPARK/Spark+Code+Style+Guide